New KickEvent PlayerEvent & Its Prefix Patch

BenjaminDaBirb opened the pull request:
Please check the line 885 in PlayerPatches.cs  for the new OnKick Prefix Patch Function, besides this also check the line 635 for the KickEvent Player Event Code.

BenjaminDaBirb committed:
Improvised Code

BenjaminDaBirb committed:
Update PlayerPatches.cs

BenjaminDaBirb committed:
New Stuff

Added OnKick Prefix Patch. Along with the KickEvent event.

BenjaminDaBirb committed:
Merge branch 'synapse3' of https://github.com/Ludens-Workshop/Synapse3 into synapse3

BenjaminDaBirb committed:
Corrected Small Error

AlmightyLks commented:
You can copy the link to those lines and GitHub will preview them

Though, the "Files Changed" Tab already gives a good overview about what was changed

BenjaminDaBirb commented:
@AlmightyLks Alright, you can view it in the Files Changed tab then.

BenjaminDaBirb commented on Synapse3.SynapseModule/Patches/PlayerPatches/PlayerPatches.cs:
```diff
@@ -53,13 +53,13 @@ public static bool OnPanelInteract(PlayerInteract __instance ,ref PlayerInteract
                 !__instance.CanInteract) return false;
 
-            var ev = new WarheadPanelInteractEvent(__instance.GetSynapsePlayer(),
+            var warheadPanelInteractEvent = new WarheadPanelInteractEvent(__instance.GetSynapsePlayer(),
```
I will change this back to the original name (ev)

BenjaminDaBirb committed:
Changed back to Ev in OnPanelInteract Patch

BenjaminDaBirb committed:
Custom Logger Utility For Synapse  3

Added a new class SynapseLogger.cs, which uses NeuronLogger by default. Designed for the sake of simplicity of course.

BenjaminDaBirb commented:
Added a Custom Logger Utility for Synapse3 in the latest commit. Utilizes the Neuron Logger but is designed for the sake of simplicity.

BenjaminDaBirb committed:
No Exception Error and Fatal Method in SynapseLogger.cs

BenjaminDaBirb committed:
Remove Exceptions and Non-Exception Methods

Just realized they were useless...

BenjaminDaBirb commented:
Removed Exception Parameter from Fatal and Error Functions, its useless.

Also removed the Non-Exception Method for Fatal and Error Functions, also useless.

GrafDimenzio commented:
The Kick Event isn't implemented properly. The Patch is not applied since it's in the wrong class and also are there 2 kick methods, so a Player could still be kicked without the event firing furthermore is a kick just a ban with the duration of 0 seconds which means the already existing ban event would fire when someone is kicked. If you really need a separate event for this, it would be easier to split the already existing ban event into 2 events with the same already existing patch.

And this Logger also has some flaws. Firstly, you doesn't store the Logger which would be quite easy and a little bit more optimized, but the main Problem with this design is that it always uses Synapse as Logging Type which Plugins and Modules should never do. When you are using the Proper calling type you instantly know which Plugin/Module logged this Info/Warning/Error and bugs can be traced back way easier. It also helps the hoster to indicate which Plugin/Module Creator they have to contact.

Also as little side note we only have 4 discord rules and you managed to broke the very first one. Please refrain from sending direct messages to me or Helight just to notify us when you created a PR. This PR is only around 15 hours old and we can't constantly work on Synapse. We live most likely in a different time zone and I got the message from Helight that you contacted him at midnight while I was almost at sleep. We have a github channel in our discord where every PR is logged, so we will see it when someone created a PR.

GrafDimenzio committed:
Generic Logger and moved Kick Patch

GrafDimenzio committed:
Split BanEvent into Kick and BanEvent

GrafDimenzio committed:
Merge branch 'synapse3' into Ludens-Workshop_synapse3

# Conflicts:
#	Synapse3.SynapseModule/Events/PlayerEvents.cs

GrafDimenzio committed:
fixed Reason setter

BenjaminDaBirb commented:
Is there anything else i need to do before this can be merged?

GrafDimenzio committed:
fixed SynapseLogger::Warn
